src/freenet/client/filter/CSSTokenizerFilter.java

```diff
@@ -4502,9 +4502,9 @@ public boolean recursiveDoubleBarVerifier(String expression,ParsedWord[] words,F
 							}
 							// Against the rest of the pattern: the part that we've tried and failed plus the part that we haven't tried yet.
 							// NOT against the verifier we were just considering, because the double-bar operator expects no more than one match from each component of the pattern.
-							String pattern = ignoredParts+((("".equals(ignoredParts))||("".equals(secondPart)))?"":"a")+secondPart;
+							String pattern = ignoredParts+(((ignoredParts.isEmpty())||(secondPart.isEmpty()))?"":"a")+secondPart;
```

Comment:
are we sure that ignoredParts and secondPart cannot be null?

Reply:
Yeah, same as above; assigned an empty string at declaration time, or assigned other values that are not `null`. :)